scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,340 @@
+---
+title: "Setup: Introduction Module"
+author: "Stephanie Spielman for the Data Lab"
+date: "`r Sys.Date()`"
+output: 
+  html_notebook:
+    toc: true
+    toc_depth: 3
+    toc_float: true
+    number_sections: true
+---
+
+```{r setup, include=FALSE}
+set.seed(2022)
+
+library(SingleCellExperiment)
+library(magrittr)
+library(SingleR) # annotation
+library(bluster) # clustering to give the people their UMAP colors
+```
+
+To get the raw data:
+```{sh eval = FALSE}
+NAME=10k_PBMCs_TotalSeq_B_3p_raw_feature_bc_matrix.tar.gz
+wget https://cf.10xgenomics.com/samples/cell-exp/6.0.0/10k_PBMCs_TotalSeq_B_3p/$NAME
+tar -xf $NAME
+# This gets unpacked into `raw_feature_bc_matrix`
+
+
+# For posterity, first attempt was with MALT, but it had wildly low ADT counts
+#NAME=malt_10k_protein_v3_raw_feature_bc_matrix.tar.gz
+#wget https://cf.10xgenomics.com/samples/cell-exp/3.0.0/malt_10k_protein_v3/$NAME
+```
+
+## Filtering and normalization
+
+```{r}
+pbmc_10x_dir <- "raw_feature_bc_matrix"
+# for now..
+mito_file <- "https://raw.githubusercontent.com/AlexsLemonade/scpca-downstream-analyses/main/reference-files/Homo_sapiens.GRCh38.104.mitogenes.txt"
+
+
+
+pbmc_sce_raw <- DropletUtils::read10xCounts(pbmc_10x_dir)
+mito_genes <- readr::read_tsv(mito_file, 
+                              col_names=FALSE) %>%
+  # only keep relevant mitochondrial genes
+  dplyr::filter(X1 %in% rownames(pbmc_sce_raw)) %>%
+  dplyr::pull(X1)
+```
+
+
+### Initial Mitochondria QC
+
+```{r}
+pbmc_sce_percellqc <- scater::addPerCellQC(
```

Comment:
> My general recommendation is going to be to not change the name of the object when adding QC stats or otherwise changing metadata, to prevent extra large copies

Normally I agree with this, but in the context of workshops specifically (with the big caveat that this workshop is likely to launch in person instead of remote) this is one of the biggest places I've seen them get tripped up. It's really easy mess up or skip one of the variable overwrites, and as a consequence participants have to back up a lot and re-run bunch of code chunks to get back to a correct/clean version of the object. The time to fix kind of problem this knocks them off course for a while since it almost always needs a breakout room. 

So, I might advocate for - using different variable names but talking about why it's a Choice, and then maybe `rm`ing on some large objects once we arrive at the filtered version? What are your thoughts on that teaching approach?